src/component.jl

```diff
@@ -868,3 +294,69 @@ function assign_new_uuid!(component::InfrastructureSystemsComponent)
     set_uuid!(get_internal(component), new_uuid)
     return
 end
+
+"""
+Attaches an info to a component
+"""
+function attach_supplemental_attribute!(
+    component::InfrastructureSystemsComponent,
+    info::T,
+) where {T <: InfrastructureSystemsSupplementalAttribute}
+    component_uuid = get_uuid(component)
+
+    if component_uuid ∈ info.components_uuid
+        throw(
+            ArgumentError(
+                "SupplementalAttribute type $T with UUID $(get_uuid(info)) already attached to component $(get_name(component))",
+            ),
+        )
+    end
+
+    push!(info.components_uuid, component_uuid)
+    infos_container = get_supplemental_attributes_container(component)
+
+    if !haskey(infos_container, T)
+        infos_container[T] = Set{T}()
```

Comment:
After thinking about what the interface should be, if we want to retrieve an attribute from the component It should be by type since the specific UUID wouldn't be known by the user like the component name.